File: apps/api/backend/schemas/review.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class ReviewCreateRequest(BaseModel):
    user_id: UUID
    product_id: UUID
    order_item_id: UUID
    rating: int = Field(ge=1, le=5)
    review_title: str = Field(min_length=1, max_length=200)
    review_content: str = Field(min_length=1, max_length=2000)


class ReviewUpdateRequest(BaseModel):
    user_id: UUID
    rating: int | None = Field(default=None, ge=1, le=5)
    review_title: str | None = Field(default=None, min_length=1, max_length=200)
    review_content: str | None = Field(default=None, min_length=1, max_length=2000)


class ReviewDeleteRequest(BaseModel):
    user_id: UUID


class ReviewResponse(BaseModel):
    review_id: UUID
    user_id: UUID
    product_id: UUID
    order_item_id: UUID
    rating: int
    review_title: str
    review_content: str | None = None
    review_status: str
    created_at: datetime
    updated_at: datetime


class ReviewCreateResponse(ReviewResponse):
    message: str


class ReviewUpdateResponse(ReviewResponse):
    message: str


class ReviewDeleteResponse(BaseModel):
    review_id: UUID
    user_id: UUID
    review_status: str
    updated_at: datetime
    message: str

File: apps/web/src/types/review.ts
export type ReviewCreateRequest = {
  user_id: string;
  product_id: string;
  order_item_id: string;
  rating: number;
  review_title: string;
  review_content: string;
};

export type ReviewCreateResponse = {
  review_id: string;
  user_id: string;
  product_id: string;
  order_item_id: string;
  rating: number;
  review_title: string;
  review_content: string;
  review_status: string;
  created_at: string;
  updated_at: string;
  message: string;
};

export type ProductReviewItem = {
  review_id: string;
  user_id: string;
  product_id: string;
  order_item_id: string;
  rating: number;
  review_title: string;
  review_content: string;
  review_status: string;
  created_at: string;
  updated_at?: string | null;
  user_name?: string | null;
};

export type ProductReviewListResponse = {
  product_id: string;
  total_reviews: number;
  average_rating?: string | number | null;
  reviews: ProductReviewItem[];
};